bundles/org.openhab.voice.googlestt/src/main/java/org/openhab/voice/googlestt/internal/GoogleSTTService.java

```diff
@@ -0,0 +1,381 @@
+        }
+
+        public void onResponse(StreamingRecognizeResponse response) {
+            lastInputTime = System.currentTimeMillis();
+            List<StreamingRecognitionResult> results = response.getResultsList();
+            logger.debug("Got {} results", response.getResultsList().size());
+            if (results.isEmpty()) {
+                logger.debug("No results");
+                return;
+            }
+            results.forEach(result -> {
+                List<SpeechRecognitionAlternative> alternatives = result.getAlternativesList();
+                logger.debug("Got {} alternatives", alternatives.size());
+                SpeechRecognitionAlternative alternative = alternatives.stream()
+                        .min(Comparator.comparing(SpeechRecognitionAlternative::getConfidence)).orElse(null);
+                if (alternative == null) {
+                    return;
+                }
+                String transcript = alternative.getTranscript();
+                logger.debug("Alternative transcript: {}", transcript);
+                logger.debug("Alternative confidence: {}", alternative.getConfidence());
+                if (result.getIsFinal()) {
+                    transcriptBuilder.append(transcript);
+                    confidenceSum += alternative.getConfidence();
+                    responseCount++;
+                    // when in single utterance mode we can just get one final result so complete
+                    if (config.singleUtteranceMode) {
+                        completeListener.accept(null);
+                    }
+                }
+            });
+        }
+
+        public void onComplete() {
+            sttListener.sttEventReceived(new SpeechStopEvent());
+            float averageConfidence = confidenceSum / (float) responseCount;
+            String transcript = transcriptBuilder.toString();
+            if (!transcript.isBlank()) {
+                sttListener.sttEventReceived(new SpeechRecognitionEvent(transcript, averageConfidence));
+            } else {
+                if (!config.noResultsMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.noResultsMessage));
+                } else {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("No results"));
+                }
+            }
+        }
+
+        public void onError(@Nullable Throwable t) {
+            logger.warn("Recognition error: ", t);
+            completeListener.accept(t);
+            sttListener.sttEventReceived(new SpeechStopEvent());
+            if (!config.errorMessage.isBlank()) {
+                sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.errorMessage));
+            } else {
+                String errorMessage = t.getMessage();
+                sttListener.sttEventReceived(
+                        new SpeechRecognitionErrorEvent(errorMessage != null ? errorMessage : "Unknown error"));
```

Comment:
Same here

Reply:
That text should never be told as the exceptions should always have a message which we can not translate. But I have to put something there because of nullability. Could we left it like that? User can also customize this.

Reply:
Ok, understood. Keep it as it is.